crates/scroll/alloy/evm/src/tx/compression.rs

```diff
@@ -0,0 +1,214 @@
+    };
+
+    /// The maximum size of the compression window in bytes (`2^CL_WINDOW_LIMIT`).
+    const CL_WINDOW_LIMIT: u32 = 22;
+
+    fn compressor(target_block_size: u32) -> Encoder<'static, Vec<u8>> {
+        let mut encoder = Encoder::new(Vec::new(), 0).expect("Failed to create zstd encoder");
+        encoder
+            .set_parameter(CParameter::LiteralCompressionMode(ParamSwitch::Disable))
+            .expect("Failed to set literal compression mode");
+        encoder
+            .set_parameter(CParameter::WindowLog(CL_WINDOW_LIMIT))
+            .expect("Failed to set window log");
+        encoder
+            .set_parameter(CParameter::TargetCBlockSize(target_block_size))
+            .expect("Failed to set target block size");
+        encoder.include_checksum(false).expect("Failed to disable checksum");
+        encoder.include_magicbytes(false).expect("Failed to disable magic bytes");
+        encoder.include_dictid(false).expect("Failed to disable dictid");
+        encoder.include_contentsize(true).expect("Failed to include content size");
+        encoder
+    }
+
+    /// Computes the compression ratio for the provided bytes.
+    ///
+    /// This is computed as:
+    /// `(original_size * TX_L1_FEE_PRECISION_U256) / compressed_size`
+    pub fn compute_compression_ratio<T: AsRef<[u8]>>(bytes: &T) -> U256 {
+        // Instantiate the compressor
+        let mut compressor = compressor(CL_WINDOW_LIMIT);
+
+        // Set the pledged source size to the length of the bytes and write the bytes to the
+        // compressor.
+        let original_bytes_len = bytes.as_ref().len();
+        compressor
+            .set_pledged_src_size(Some(original_bytes_len as u64))
+            .expect("failed to set pledged source size");
+        compressor.write_all(bytes.as_ref()).expect("failed to write bytes to compressor");
+
+        // Finish the compression and get the result.
+        let result = compressor.finish().expect("failed to finish compression");
+
+        // compute the compression ratio
+        let original_len = U256::from(original_bytes_len).saturating_mul(TX_L1_FEE_PRECISION_U256);
+        let compressed_len = U256::from(result.len());
+        original_len.wrapping_div(compressed_len)
+    }
+}
+
+#[cfg(not(feature = "zstd_compression"))]
+mod zstd_compression {
+    use super::*;
+
+    /// Computes the compression ratio for the provided bytes. This panics if the compression
+    /// feature is not enabled. This is to support `no_std` environments where zstd is not
+    /// available.
+    pub fn compute_compression_ratio<T: AsRef<[u8]>>(_bytes: &T) -> U256 {
+        panic!("Compression feature is not enabled. Please enable the 'compression' feature to use this function.");
+    }
+}
+
+/// A generic wrapper for a type that includes a compression ratio and encoded bytes.
+#[derive(Debug, Clone)]
+pub struct WithCompression<T> {
+    value: T,
+    compression_ratio: U256,
+    encoded_bytes: Bytes,
```

Comment:
```suggestion
    rlp_bytes: Bytes,
```

The original name feels to me as if it was the compressed tx bytes, while it is not.